### PR TITLE
mgr/nfs: validate virtual_ip parameter

### PR DESCRIPTION
**Pull request by rkachach**
With the fix:

```
[ceph: root@ceph-node-0 /]# ceph nfs cluster create foo --ingress --virtual-ip 192.168.111/24 --port 2049
Error EPERM: '192.168.111' does not appear to be an IPv4 or IPv6 address

```

Fixes: https://tracker.ceph.com/issues/54581
Signed-off-by: Redouane Kachach <rkachach@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/nfs/cluster.py [hunk 1]
```diff
@@ -105,7 +106,11 @@ def create_nfs_cluster(
             ingress: Optional[bool] = None,
             port: Optional[int] = None,
     ) -> Tuple[int, str, str]:
+
         try:
+            if virtual_ip:
+                ip = virtual_ip.split('/')[0]
+                ipaddress.ip_address(ip)
```
Comment (phlogistonjohn) [inline src/pybind/mgr/nfs/cluster.py:113]: I assume that this relies on the ip_address function raising an exception. That's good, but maybe a short comment on stuff like this in the future helps readers understand that you're relying on the exception and  the return value of the function is being discarded on purpose. But it's not a big deal. :-)

**Review by phlogistonjohn — APPROVED** [review]
LGTM

File: src/pybind/mgr/nfs/cluster.py (hunk 1 above)
Comment (rkachach) [reply src/pybind/mgr/nfs/cluster.py:113]: yeah, your are right, I'll add a comment to clarify what is the code is trying to validate :)

**Comment by rkachach** [comment]
jenkins test api

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
jenkins test api

**Pull request merged**